CONTRIBUTING.md

```diff
@@ -0,0 +1,192 @@
+
+## Contributing by adding an image:
+
+In case you would like to build a new container image that does not exist yet in our registry, then you have to send a _Merge Request_ to the [kubic automation repo](https://github.com/kubic-project/automation) in order to activate add the instructions for the Concourse pipeline to take this new image of yours into account. Otherwise, you have to manually create a _SubmitRequest_ to the appropriate _devel project_ in OBS. To highlight this for the reviewer, a proposed label for this should be `new_image`.
+
+## Test before you submit
+
+### Fetch the code from OBS
+
+Nowhere in this process there is any testing automation. Hopefully, since you are building and modifying these images, you should be able to test your own changes by running some local tests. To do that, you can build your container image locally using [kiwi image building system](https://doc.opensuse.org/projects/kiwi/doc/). To do that you have to perform a checkout of the image you want from OBS. Then inside that project/branch you have, you can just modify the _.kiwi_ file accordingly. When your image is building in the OBS, it uses [pre_checking.sh script](https://github.com/kubic-project/container-images/blob/master/pre_checkin.sh), but you __should not use it__ when you are testing locally. This is because the project in OBS is templated in a way that certain _placeholders_ are replaced by values taken from GitHub -- that means that it will accidentally overwrite your local changes.
+
+> __WARNING__: You should NOT use this `pre_checking.sh` for testing locally
+
+### Templated structure
+
+As soon as you have _cloned/checked-out_ the code from OBS to your local dis, then it's time to build the container image. But before doing that it would be wise to be informed first about an existing a bug in the `osc` tool that does not trigger correctly the `replace_using_package_version` service. This is an OBS service that ensures that the image we are building has the same tag with the value of the OBS package version that provides. To understand this better, let's look for example at the [kubic-caasp-dex-image](https://build.opensuse.org/package/show/devel:CaaSP:kubic-container/kubic-caasp-dex-image).
+
+```bash
+_service # Synchronisation between the tag of the container image with the RPG pkg version
+_servicedata # Links thie GH repo with OBS devel repo of the image
```

Comment:
This is not needed any more

Reply:
Correct, this is not strictly needed anymore. The `_servicedata` was used to autogenerate the changes file with the `obs_scm` service, but this turned to be not flexible enough.